rcode-reader/src/main/java/eu/rssw/pct/elements/DatasetElement.java

```diff
@@ -8,38 +27,44 @@
 import eu.rssw.pct.RCodeInfo;
 
 public class DatasetElement extends AbstractAccessibleElement {
-  protected int prvte;
-  protected int flags;
-  protected int crc;
+  private final int prvte;
+  private final String[] bufferNames;
+  private final DataRelationElement[] relations;
 
-  private String[] bufferNames;
-  protected DataRelationElement[] relations;
-
-  public DatasetElement(String name, Set<AccessType> accessType, byte[] segment, int currentPos, int textAreaOffset, ByteOrder order) {
+  public DatasetElement(String name, Set<AccessType> accessType, int prvte, String[] bufferNames,
+      DataRelationElement[] relations) {
     super(name, accessType);
+    this.prvte = prvte;
+    this.bufferNames = bufferNames;
+    this.relations = relations;
+  }
 
+  public static DatasetElement fromDebugSegment(String name, Set<AccessType> accessType, byte[] segment, int currentPos,
+      int textAreaOffset, ByteOrder order) {
     int bufferCount = ByteBuffer.wrap(segment, currentPos, Short.BYTES).order(order).getShort();
     int relationshipCount = ByteBuffer.wrap(segment, currentPos + 2, Short.BYTES).order(order).getShort();
-    this.prvte = ByteBuffer.wrap(segment, currentPos + 4, Short.BYTES).order(order).getShort();
-    this.flags = ByteBuffer.wrap(segment, currentPos + 6, Short.BYTES).order(order).getShort();
-    this.crc = ByteBuffer.wrap(segment, currentPos + 8, Short.BYTES).order(order).getShort();
+    int prvte = ByteBuffer.wrap(segment, currentPos + 4, Short.BYTES).order(order).getShort();
+    // int flags = ByteBuffer.wrap(segment, currentPos + 6, Short.BYTES).order(order).getShort();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ACommentedOutCodeLine)